docs/src/main/asciidoc/spring-cloud-commons.adoc

```diff
@@ -1059,3 +1059,99 @@ myrandom=${cachedrandom.appname.value}
 == Configuration Properties
 
 To see the list of all Spring Cloud Commons related configuration properties please check link:appendix.html[the Appendix page].
+
+== Spring Cloud Request Hedging
+=== Introduction
+Request hedging is a mechanism to improve performance of idempotent requests with large worst-case response times. It
+does this by automatically retrying after the original request has been active for some configurable amount of time,
+usually based on a quantile of the request's typical response time. Unlike timing out the request and retrying, request
+hedging will keep the original requests and return the first request to succeed. This trades off a small percentage
+of additional network traffic to reduce the long-tail latency.
+
+=== Usage
+You can configure a `WebClient` to do request hedging by creating a `HedgingClient` `@Bean` and use the `@Hedged`
+qualifier. The `HedgingClient` allows you to specify how long to wait before executing the extra requests, a maximum
+number of extra requests to create. Most users will want the `LatencyPercentileHedgingClient`, which executes new
+requests when the previous ones have exceeded some percentile of the historical latency.
+
+This example shows a typical usage:
+
+====
+[source,java,indent=0]
+----
+@Configuration
+public class MyConfiguration {
+    @Bean
+    HedgingClient hedgingClient() {
+    	return new LatencyPercentileHedgingClient(
+            3, // Maximum extra requests to execute
```

Comment:
Suggestion: let's create a builder for `LatencyPercentileHedgingClient` so that these comments can turn into self-documenting setter methods.

Reply:
That feels like too much complication for this thing. I'm not sure this client is the right way to do it anyway.